modules/ethereum-contract/builtin/src/lib.rs

```diff
@@ -113,16 +131,35 @@ pub fn verify_substrate_finality_proof(
 ) -> Result<(), Error> {
 	let best_set = AuthorityList::decode(&mut &raw_best_set[..])
 		.map_err(Error::BestSetDecode)
-		.and_then(|authorities| VoterSet::new(authorities.into_iter()).ok_or(Error::InvalidBestSet))?;
-	sc_finality_grandpa::GrandpaJustification::<Block>::decode_and_verify_finalizes(
+		.and_then(|authorities| VoterSet::new(authorities.into_iter()).ok_or(Error::InvalidBestSet));
+
+	log::trace!(
+		target: "bridge-builtin",
+		"Parsed Substrate authorities set {}: {:?}",
+		hex::encode(raw_best_set),
+		best_set,
+	);
+
+	let best_set = best_set?;
+
+	let verify_result = sc_finality_grandpa::GrandpaJustification::<Block>::decode_and_verify_finalizes(
 		&raw_finality_proof,
 		(finality_target_hash, finality_target_number),
 		best_set_id,
 		&best_set,
 	)
 	.map_err(Box::new)
 	.map_err(Error::JustificationVerify)
-	.map(|_| ())
+	.map(|_| ());
+
+	log::debug!(
+		target: "bridge-builtin",
+		"Verified Substrate finality proof {}: {:?}",
+		hex::encode(raw_finality_proof),
```

Comment:
Maybe full hex value should only be displayed if we are `tracing`? I supposed the output would be cleaner if we just do `0x123...beef` and display the decoded values and to see the actual hex we enable `trace`, wdyt?

Reply:
Ok, now it is only printed when trace is enabled